examples/Mass Groups & Collaborations Update/Mass_Groups_Collabs_Update.ps1

```diff
@@ -0,0 +1,438 @@
+        $CurrentFolderID = $CollaborationInputEntry.$FolderIdColumnName
+
+        # Ensure that it's not empty
+        if (!$CurrentFolderID) {
+            Write-Log "'$FolderIdColumnName' column in row $currentRowIndex) contains invalid empty value. Skipping processing current row." -errorMessage "Invalid input data." -output true -color Red
+            continue
+        }
+
+        # Get the name of the collaboration role being processed in the current row
+        $CurrentCollaborationRole = $CollaborationInputEntry.$CollaborationRoleColumnName
+
+        if (!$CurrentCollaborationRole) {
+            # Use default collaboration role if it's empty
+            $CurrentCollaborationRole = $DefaultCollaborationRole
+        }
+
+        # Ensure that collaboration role is set to valid value
+        if (!$AvailableCollaborationRoles.Contains($CurrentCollaborationRole)) {
+            Write-Log "'$CollaborationRoleColumnName' column in row $currentRowIndex) contains invalid value: '$CurrentCollaborationRole'. Please use one of these: '$($AvailableCollaborationRoles -join ", ")'. Skipping processing current row." -errorMessage "Invalid input data." -output true -color Red
+            continue
+        }
+
+        # If current processing folderId is not in FolderGroupsCollaborationHashtable, then fetch collaborations of this folder
+        if (!$FolderGroupsCollaborationHashtable.ContainsKey($CurrentFolderID)) {
+            try {
+                # Get members of the group
+                $FolderCollaborationsResp = "$(box folders:collaborations $CurrentFolderID --fields='accessible_by' --json 2>&1)"
+                $FolderCollaborations = $FolderCollaborationsResp | ConvertFrom-Json
+
+                # Update hashtable FolderGroupsCollaborationHashtable by adding list of group ids to particular group id
+                $FolderGroupsCollaborationHashtable.$CurrentFolderID = @()
+                $FolderCollaborations | Where-Object { $_.accessible_by.type -eq 'group' } | ForEach-Object { $FolderGroupsCollaborationHashtable.$CurrentFolderID += $_.accessible_by.id }
+
+                Write-Log "Successfully fetched the collaborations of the folder with ID: $CurrentFolderID." -output true
+                Write-Log $FolderCollaborations
+            } catch {
+                Write-Log "Could not fetch the collaborations of the folder with ID: $CurrentFolderID. See log for details. Skipping processing current row." -errorMessage $FolderCollaborationsResp -output true -color Red
+                continue
+            }
+        }
+
+        # Create new group collaboration to folder only if it's not there already
+        if($FolderGroupsCollaborationHashtable[$CurrentFolderID].Contains($CurrentGroupID)) {
+            Write-Log "Group '$CurrentGroupName' (ID:$CurrentGroupID) is already a collaborator of folder ID: $CurrentFolderID. Skipping processing current row." -output true
```

Comment:
Collaborations Creation: If I understand it correct, the  script works in a way that if there was no collaboration defined for a given file and given group, that it will be created. But if let's say, collaboration was there and you just want to change the collaboration role, the script won't do anything.

Do you think we can do extra check if collaboration role changed and update that if we detect that CSV defines new role?